src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -171,7 +171,6 @@
           ],
           "relatedLocations": [
             {
-              "message": "\"count\" was declared here.",
```

Comment:
Changing the type of `result.relatedLocations` from `annotatedCodeLocation[]` to `location[]` is indeed part of the larger location redesign (oasis-tcs/sarif-spec#130). We didn't originally envision that change as part of this PR, but it's fine if you do it.

But to do that, you have to give the `location` object the necessary features of the `annotatedCodeLocation` object. That means it needs a `message` property. Now if you look at the [Schema changes](https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/Schema%20changes%20between%20SARIF%20v1%20and%20v2.md) . document, you'll see that `location.message` should be of type `message`, not `string`. But we haven't yet introduced the `message` object; that's part of oasis-tcs/sarif-spec#84 (localization). So declare `location.message` as `string` for now.

Reply:
Ah, I just saw your comment acknowledging the unit test gap. Our plan of record is now to keep the unit tests up to date as we go, so that needs to be part of this PR.